### ISSUE TITLE
Request for support "Lenovo Ideapad Slim 3i Chromebook Plus"

### ISSUE THREAD

**Issue by micheleangelofabbri** [MrChromebox/firmware#643]
Hello all,

I somewhat recently picked up a "Lenovo Ideapad Slim 3i Chromebook **Plus**" on the cheap during a recent sale and decided to see if I could get Debian or PopOS running on it. I had great success with the entire process up until the reboot after installing Linux. The problem I have sounds incredibly similar to the issue laid out in the following post - [Flashed Lenovo Ideapad Slim 3i Chromebook Now Unrecognized Platform - Unable to Update](https://forum.chrultrabook.com/t/flashed-lenovo-ideapad-slim-3i-chromebook-now-unrecognized-platform-unable-to-update/974)

It looks like [s1gnate](https://forum.chrultrabook.com/u/s1gnate) and [chipmonger](https://forum.chrultrabook.com/u/chipmonger) were able to have some success with it. For me:

- Both Debian and PopOS will randomly work or not work on boot
- Sometimes I get crc errors during the boot process and sometimes I don’t
- Coreboot bios screen reads 0MB of RAM
- My board name according to chrome://version/ is `pujjoteen15w_cbx`
- In the VT-2 and Recovery Mode screen my board name comes up as `PUJJOTEEN-JQLW`
- When the MrChromebox script gets run, it tells me that my board name is `PUJJOTEEN-JQLW`

I’m at a bit of a loss at this point. Is the disparity in the board name something to be concerned with or might it be something completely unrelated? Looking over the list of supported devices, I do not see the **Plus** version of this devices present. 

I’m a bit out of my element when it comes to the differences in these boards and UEFI stuff. Could anyone point me in the right direction for further troubleshooting?

**Comment by MrChromebox** [comment]
> Looking over the list of supported devices, I do not see the Plus version of this devices present.

because, as emphasized at the top of the supported devices list, the make/model are irrelevant. And trying to list every marketing name variation of a device would be exhausting.

The only thing that matters is that the firmware flashed to your device is built from the correct mainboard profile. In most cases, this is a straightforward 1:1 mapping. It would seem not to be here though -- your HWID is PUJJOTEEN, but the board name is pujjoteen15w (which is a slightly different board). The easiest way to fix this is likely for you to go back to stock ChromeOS, then re-flash the firmware once I've adjusted the script HWID->firmware image mapping.

**Comment by micheleangelofabbri** [comment]
Thank you for the quick response and all the incredible work on this project. I've already gone back to the stock ChromeOS and eagerly await your updates.

**Comment by MrChromebox** [comment]
I've updated the script so it should flash the pujjoteen15w firmware now - you can verify by the filename being downloaded when flashing. hopefully that resolves your issues.

**Comment by micheleangelofabbri** [comment]
Amazing, I'll give it a whirl. I'll update tomorrow on my success/failure.

**Comment by micheleangelofabbri** [comment]
I ran the updated script and got the pujjoteen15w image this time. Unfortunately I'm still running into similar problems with the boot to debian or popos only occasionally working. I've tried with and without secure boot to see if that made a difference. Something else worth mentioning that might be a concern is that this machine has an internal eMMC device for the hard drive. I'm not sure if that would affect anything though. Booting from flash drives works flawlessly.

![PXL_20240531_113249362](https://github.com/MrChromebox/firmware/assets/95646303/4f4dd36d-9a06-4a84-af76-0395921cbd74)
![PXL_20240531_131010907](https://github.com/MrChromebox/firmware/assets/95646303/85680954-82e4-4963-be0a-b34ad5e048bd)

Thanks again for the time and attention on this problem.

**Comment by MrChromebox** [comment]
Does the device still show 0MB RAM on the firmware setup screen?
Are you able to boot a live Linux USB and get a cbmem log?

`wget https://mrchromebox.tech/files/util/cbmem.tar.gz`
`tar zxf cbmem.tar.gz`
`sudo ./cbmem -1 > cbmem.log`

attach log here

**Comment by micheleangelofabbri** [comment]
Sadly, the coreboot screen does still read 0MB RAM. Log file has been attached 🪵

[cbmem.log](https://github.com/user-attachments/files/15519090/cbmem.log)

**Comment by MrChromebox** [comment]
how about the outputs of `sudo dmidecode > dmidecode.log` and `sudo dmesg > dmesg.log` -- thx

**Comment by micheleangelofabbri** [comment]
More logs 🪵🪵
[dmesg.log](https://github.com/user-attachments/files/15520875/dmesg.log)
[dmidecode.log](https://github.com/user-attachments/files/15520876/dmidecode.log)

**Comment by micheleangelofabbri** [comment]
@MrChromebox just checking in to see if there is anything else I should be doing on my end to help troubleshoot. Thank you again for the time and effort you've put into this project already <3

**Comment by MrChromebox** [comment]
dmidecode shows that coreboot/FSP is failing to generate a valid table with the memory info, which is why the reported RAM is 0MB. I'd like to give a newer coreboot/FSP build a try, shoot me an email or ping me on Discord and I'll give you a link to a 2405 test build to try

**Comment by micheleangelofabbri** [comment]
I was able to flash with the test fw-util that was provided via email.

We are now successfully reading the correct amount of RAM in coreboot 🥳
![PXL_20240606_204240990](https://github.com/MrChromebox/firmware/assets/95646303/d686dc4d-7cba-48bb-bc52-bba02b5fdc79)

And here is the cbmem.log as well.
[cbmem_2024-06-06.log](https://github.com/user-attachments/files/15686410/cbmem_2024-06-06.log)

Unfortunately, I am still running into the same boot issues when attempting to get into Debian. Occasionally it boots correctly, but most of the time I'm getting the same boot errors. I tried a fresh install of Debian as well just in case that would help anything, but alas the same problems persist.

**Comment by MrChromebox** [comment]
this sounds like some eMMC tuning is needed in edk2, which I'm guessing Google does in depthcharge for ChromeOS.
I assume you have the same issue when booting via RW_LEGACY  firmware?

**Comment by micheleangelofabbri** [comment]
I had not tried to use the RW_LEGACY firmware, but I can certainly explore that option if you think it is a worthwhile pursuit. 

Looking though a bit of the website, it sounds as though I will have to re-enable firmware write-protect for the utility script to give me the option to use RW_LEGACY. Correct?

**Comment by MrChromebox** [comment]
> I had not tried to use the RW_LEGACY firmware, but I can certainly explore that option if you think it is a worthwhile pursuit.

not sure it's worth going back to stock for

> Looking through a bit of the website, it sounds as though I will have to re-enable firmware write-protect for the utility script to give me the option to use RW_LEGACY. Correct?

not at all. You just need to restore the stock firmware, exit/-re-run the script, then flash the RW_LEGACY firmware (and set the firmware boot flags to legacy boot + long delay)

**Comment by jmccrohan** [comment]
Hi @MrChromebox just a quick note to say thanks for all your work on this.

I installed Debian Testing last night on my `PUJJOTEEN-JQLW` and all working very smoothly now.

For reasons unknown I received an "failed: verify it contains a 64-bit UEFI OS" error on reboot after running Debian (net)installer. The EFI files were there on boot partition and `debian` entry was present in coreboot menu.

In any case, I resolved the issue after using netinstaller to chroot into the root and boot partitions and reinstall GRUB2.

Thanks again!
